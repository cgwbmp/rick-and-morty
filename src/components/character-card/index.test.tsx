import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CharacterCard from '.';

describe('<CharacterCard />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CharacterCard name="Name" image="#" />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('DOM structure not changed', () => {
    const component = renderer.create(<CharacterCard name="Name" image="#" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
