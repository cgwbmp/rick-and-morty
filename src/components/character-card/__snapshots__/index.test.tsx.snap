// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterCard /> DOM structure not changed 1`] = `
<div
  className="card"
>
  <span
    className="image"
    style={
      Object {
        "display": "inline-block",
        "height": 0,
        "width": 0,
      }
    }
  />
  <h4
    className="title"
  >
    Name
  </h4>
</div>
`;
